### PR TITLE
Param: Type List Syntax

### PR DESCRIPTION
**Pull request by ax3l**
Use a more natural `using Sequence = MakeSeq_t< 1, 2, 3 >;` syntax for user input of type lists via `MakeSeq_t` and C++11 `using`.

Avoids boiler plate code without `=` and additional `::type` access:
`typedef Seq< 1, 2, 3 >::type List;`

Cleaned up syntax of `speciesDefinition` on the way.

### CODE REVIEW

File: src/picongpu/include/simulation_defines/param/speciesDefinition.param [hunk 1]
```diff
@@ -169,52 +154,52 @@ struct Hydrogen
...
+>;
 
 /*define specie ions*/
-typedef Particles<
+using PIC_Ions = Particles<
     ParticleDescription<
-        bmpl::string<'i'>,
+        bmpl::string< 'i' >,
         SuperCellSize,
-        DefaultAttributesSeq,
```
Comment (ax3l) [inline src/picongpu/include/simulation_defines/param/speciesDefinition.param]: ha, nice bug in `PluginController.hpp`: radiation assumes *all species* have momentumPrev1.

Comment (psychocoderHPC) [inline src/picongpu/include/simulation_defines/param/speciesDefinition.param]: Right the reason is that the pusher extention can not differ between species with and without `momentumPrev1`.
To change this wee need to cleanup the pusher extension first.

Comment (ax3l) [inline src/picongpu/include/simulation_defines/param/speciesDefinition.param]: yep, will keep the "oh, ions also need all radiation flags" setting for now

File: examples/SingleParticleTest/include/simulation_defines/param/speciesDefinition.param [hunk 2]
```diff
@@ -90,76 +73,24 @@ typedef bmpl::vector<
...
 
 /*########################### end species ####################################*/
 
-
-/*! we delete this ugly definition of VectorAllSpecies after all picongpu components
- * support multi species */
-/** \todo: not nice, but this should be changed in the future*/
-typedef MakeSeq<
+using VectorAllSpecies = MakeSeq_t<
+    MakeSeq_t<
```
Comment (psychocoderHPC) [inline examples/SingleParticleTest/include/simulation_defines/param/speciesDefinition.param]: Please remove one `MakeSeq_t`

Comment (ax3l) [inline examples/SingleParticleTest/include/simulation_defines/param/speciesDefinition.param]: hups, done!

**Pull request merged**